apps/frontend/src/pages/Send/components/TxUtxoDetails.tsx

```diff
@@ -1,74 +1,126 @@
-import React, { useContext } from 'react';
+import { useState } from 'react';
 import { Psbt } from 'bitcoinjs-lib';
 import { Buffer } from 'buffer';
+import { AdjustmentsIcon, XIcon } from '@heroicons/react/outline';
 
 import { Unit, Price } from 'src/components';
 
 import { createUtxoMapFromUtxoArray, getFee } from 'src/utils/send';
 import { cloneBuffer } from 'src/utils/other';
 
-import { AccountMapContext } from 'src/context';
+import { LilyOnchainAccount, UtxoMap, UTXO } from '@lily/types';
 
-import { LilyOnchainAccount, UtxoMap } from '@lily/types';
+import { SelectInputsForm } from './SelectInputsForm';
 
 interface Props {
   currentAccount: LilyOnchainAccount;
   psbt: Psbt;
+  adjustInputs?: (utxos: UTXO[]) => Promise<Psbt>;
+  closeModal: () => void;
 }
 
-const TransactionUtxoDetails = ({ currentAccount, psbt }: Props) => {
+const TransactionUtxoDetails = ({ currentAccount, psbt, adjustInputs, closeModal }: Props) => {
+  const [showSelectInputsForm, setShowSelectInputsForm] = useState(false);
+
   const { availableUtxos, transactions } = currentAccount;
+
   const _fee = getFee(psbt, transactions);
   let utxosMap: UtxoMap;
   if (availableUtxos) {
     utxosMap = createUtxoMapFromUtxoArray(availableUtxos);
   }
 
+  if (showSelectInputsForm) {
+    return (
+      <SelectInputsForm
+        onSave={(utxos: UTXO[]) => {
+          if (adjustInputs) {
+            adjustInputs(utxos);
+            setShowSelectInputsForm(false);
+          }
+        }}
+        cancel={() => setShowSelectInputsForm(false)}
+        currentAccount={currentAccount}
+        requiredSendAmount={psbt.txOutputs[0].value}
```

Comment:
This assumes a two output transaction (index 0 = intended recipient, index 1 = change). Might not always be the case.